### PR TITLE
chore(issuer-api): Unify certificate and certification request IDs

### PR DESCRIPTION
**Pull request by josipbagaric**
- [x] Use `certificate.tokenId` as `certificate.id` and remove tokenId
- [x] Use `certificationRequest.requestId` as `certificationRequest.id` and remove requestId
- [x] Make sure migrations work (tested on a cloned database from canary - works fine)
- [ ] ~Handle cases where there is a possibility of nonce conflicts~ (Will create a new PR for this)

### CODE REVIEW

File: packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        await queryRunner.query(`DROP SEQUENCE "issuer_certification_request_id_seq"`);
+
+        await Promise.all(
+            oldCertificationRequests.map(async (oldCertReq: any) => {
+                return queryRunner.query(
+                    `UPDATE "issuer_certification_request" SET id = ${oldCertReq.requestId} WHERE id = '${oldCertReq.id}'`
+                );
+            })
+        );
+    }
```
Comment (ioncreature) [inline packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts]: Dangerous approach, we can lose all `tokenId`'s and down migration won't help.
Let's better first copy `tokenId`'s to 'id' and then drop `tokenId` column

File: packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            `ALTER TABLE "issuer_certification_request" ADD CONSTRAINT "UQ_551869cc9ee5caeccd53c966cdd" UNIQUE ("requestId")`
+        );
+
+        await Promise.all(
+            oldCertificationRequests.map(async (oldCertReq: any) => {
+                return queryRunner.query(
+                    `UPDATE "issuer_certification_request" SET "requestId" = ${oldCertReq.requestId} WHERE id = '${oldCertReq.id}'`
+                );
+            })
+        );
```
Comment (ioncreature) [inline packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts]: same as above

File: packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts [hunk 3]
```diff
@@ -5,17 +5,17 @@ import { createReadStream } from 'fs';
...
+
 import { CreateIrecCertificationRequestCommand } from '../commands';
 import { IrecCertificateService } from '../irec-certificate.service';
 import { FullCertificationRequestDTO } from '../full-certification-request.dto';
 import { IrecCertificationRequest } from '../irec-certification-request.entity';
 
 @CommandHandler(CreateIrecCertificationRequestCommand)
-export class CreateCertificationRequestHandler
-    extends OriginalHandler
+export class CreateIrecCertificationRequestHandler
```
Comment (ioncreature) [inline packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts:17]: Why did you change the name?

Comment (ioncreature) [reply packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts:17]: It probably wont work with `origin-irec-backend`

**Review by ioncreature — CHANGES_REQUESTED** [review]

File: packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts (hunk 1 above)
Comment (soanvig) [reply packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts]: We should consider not dropping or rewriting columns at all, but just rename columns around and in future remove old ones. In this case this would rename `id` to `id_bak`, then rename `requestId` to `id`

File: packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts (hunk 2 above)
Comment (soanvig) [reply packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts]: and the same as above in my case as well

File: packages/traceability/issuer-api/migrations/1622459627494-RmCertificationRequestStatus.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class RmCertificationRequestStatus1622459627494 implements MigrationInterface {
+    name = 'RmCertificationRequestStatus1622459627494';
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "issuer_certification_request" DROP COLUMN "status"`);
```
Comment (soanvig) [inline packages/traceability/issuer-api/migrations/1622459627494-RmCertificationRequestStatus.ts:7]: This is some additional change?

File: packages/traceability/issuer-api/src/pods/certification-request/certification-request.controller.ts [hunk 5]
```diff
@@ -122,7 +122,7 @@ export class CertificationRequestController {
             new ValidateCertificationRequestCommand(dto)
         );
 
-        if (!validationCheck.success) {
+        if (!validationCheck?.success) {
             return validationCheck;
```
Comment (soanvig) [inline packages/traceability/issuer-api/src/pods/certification-request/certification-request.controller.ts]: This code doesn't make sense now - if validationCheck can be undefined, and THEN we enter the if, the undefined will be returned

**Review by soanvig — CHANGES_REQUESTED** [review]

File: packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts (hunk 1 above)
Comment (josipbagaric) [reply packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts]: Good point @ioncreature. It's an edge case, since tokenIds are stored in memory, but a valid one.

@soanvig I would consider this only in cases where there's no clear `down()` migrations. Here it's unnecessary, as the `down()` migration will work seamlessly with existing data.

File: packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts (hunk 3 above)
Comment (josipbagaric) [reply packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts:17]: Changed the name because it works differently now - it calls the original handler and supplements it with I-REC features. It doesn't override the original handler.

Why won't it work? I'll do more thorough testing on this.

File: packages/traceability/issuer-api/src/pods/certification-request/certification-request.controller.ts (hunk 5 above)
Comment (josipbagaric) [reply packages/traceability/issuer-api/src/pods/certification-request/certification-request.controller.ts]: Good point, didn't pay too much attention to this just wanted it to compile. Will rework this part.

File: packages/traceability/issuer-api/migrations/1622459627494-RmCertificationRequestStatus.ts (hunk 4 above)
Comment (josipbagaric) [reply packages/traceability/issuer-api/migrations/1622459627494-RmCertificationRequestStatus.ts:7]: Yup. Certification requests no longer need the `status` field because they are stored in the db only once they are deployed to the blockchain (they are no longer queued - this queue will be replaced with an application wide-queue).

File: packages/traceability/issuer-api/src/pods/certification-request/certification-request.controller.ts (hunk 5 above)
Comment (josipbagaric) [reply packages/traceability/issuer-api/src/pods/certification-request/certification-request.controller.ts]: Undid the change for now, as it's not needed it seems. Probably pushed by mistake.

File: packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts (hunk 3 above)
Comment (ioncreature) [reply packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts:17]: It could not work because this package is like a proxy for `issuer-api`. And `origin-backend-irec-app` works only with this package, and not with `issuer-api` anymore. 
So since this package is a proxy, it has to have the same signatures with `issuer-api` everywhere.

File: packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        await queryRunner.query(`ALTER TABLE "issuer_certificate" ALTER COLUMN "id" DROP DEFAULT`);
+        await queryRunner.query(`DROP SEQUENCE "issuer_certificate_id_seq"`);
+
+        await Promise.all(
+            oldCertificates.map(async (oldCertificate: any) => {
+                return queryRunner.query(
+                    `UPDATE "issuer_certificate" SET id = ${oldCertificate.tokenId} WHERE id = '${oldCertificate.id}'`
+                );
+            })
+        );
```
Comment (ioncreature) [inline packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts]: 1. Let's do `DROP SEQUENCE IF EXISTS "issuer_certificate_id_seq"`. This is for the case when we run the same migration more than once (eg it failed on the first run)
2. `await Promise.all()` looks like we are ddosing our own DB. Let's rewrite it with `for` loop or with longer SQL query or with batch SQL query

**Review by ioncreature — CHANGES_REQUESTED** [review]

File: packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts (hunk 3 above)
Comment (josipbagaric) [reply packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts:17]: How do I test this flow? I see no automatic tests in the package.

Comment (ioncreature) [reply packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts:17]: Manually only, I'll add tests later

**Review by soanvig — APPROVED** [review]
Don't trust my approval too much though

File: packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts (hunk 3 above)
Comment (ioncreature) [reply packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/create-irec-certification-request.handler.ts:17]: added tests

**Review by ioncreature — APPROVED** [review]
